Index: perl/files/eblits/src_prepare-v50160001.eblit
===================================================================
--- perl.orig/files/eblits/src_prepare-v50160001.eblit
+++ perl/files/eblits/src_prepare-v50160001.eblit
@@ -57,6 +57,8 @@ eblit-perl-src_prepare() {
 		EPATCH_SINGLE_MSG="  ${patch} ..."
 		epatch "${WORKDIR}"/patches/${patch}
 	done < "${WORKDIR}"/patches/series
+	epatch "${FILESDIR}"/${PN}-cwd-prefix.patch
+	eprefixify dist/Cwd/Cwd.pm
 
 	src_prepare_update_patchlevel_h
 
