Index: perl/files/eblits/src_configure-v50160001.eblit
===================================================================
--- perl.orig/files/eblits/src_configure-v50160001.eblit
+++ perl/files/eblits/src_configure-v50160001.eblit
@@ -141,5 +141,6 @@ eblit-perl-src_configure() {
 		-Dinstallusrbinperl='n' \
 		-Ud_csh \
 		-Uusenm \
+		-Dsh="${EPREFIX}"/bin/sh \
 		"${myconf[@]}" || die "Unable to configure"
 }
