Index: portage-2.2.28/pym/portage/package/ebuild/prepare_build_dirs.py
===================================================================
--- portage-2.2.28.orig/pym/portage/package/ebuild/prepare_build_dirs.py
+++ portage-2.2.28/pym/portage/package/ebuild/prepare_build_dirs.py
@@ -153,12 +153,12 @@ def _prepare_features_dirs(mysettings):
 
 	features_dirs = {
 		"ccache":{
-			"path_dir": "/usr/%s/ccache/bin" % (libdir,),
+			"path_dir": "%s/usr/%s/ccache/bin" % (mysettings["EPREFIX"], libdir),
 			"basedir_var":"CCACHE_DIR",
 			"default_dir":os.path.join(mysettings["PORTAGE_TMPDIR"], "ccache"),
 			"always_recurse":False},
 		"distcc":{
-			"path_dir": "/usr/%s/distcc/bin" % (libdir,),
+			"path_dir": "%s/usr/%s/distcc/bin" % (mysettings["EPREFIX"], libdir),
 			"basedir_var":"DISTCC_DIR",
 			"default_dir":os.path.join(mysettings["BUILD_PREFIX"], ".distcc"),
 			"subdirs":("lock", "state"),
