Index: gcc-4.9.2/gcc/config/i386/linux64.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/i386/linux64.h
+++ gcc-4.9.2/gcc/config/i386/linux64.h
@@ -27,6 +27,6 @@ see the files COPYING3 and COPYING.RUNTI
 #define GNU_USER_LINK_EMULATION64 "elf_x86_64"
 #define GNU_USER_LINK_EMULATIONX32 "elf32_x86_64"
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux-x86-64.so.2"
-#define GLIBC_DYNAMIC_LINKERX32 "/libx32/ld-linux-x32.so.2"
+#define GLIBC_DYNAMIC_LINKER32 DYNAMIC_LINKER_PREFIX "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER64 DYNAMIC_LINKER_PREFIX "/lib64/ld-linux-x86-64.so.2"
+#define GLIBC_DYNAMIC_LINKERX32 DYNAMIC_LINKER_PREFIX "/libx32/ld-linux-x32.so.2"
Index: gcc-4.9.2/gcc/config/arm/linux-eabi.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/arm/linux-eabi.h
+++ gcc-4.9.2/gcc/config/arm/linux-eabi.h
@@ -68,8 +68,8 @@
    GLIBC_DYNAMIC_LINKER_DEFAULT and TARGET_DEFAULT_FLOAT_ABI.  */
 
 #undef  GLIBC_DYNAMIC_LINKER
-#define GLIBC_DYNAMIC_LINKER_SOFT_FLOAT "/lib/ld-linux.so.3"
-#define GLIBC_DYNAMIC_LINKER_HARD_FLOAT "/lib/ld-linux-armhf.so.3"
+#define GLIBC_DYNAMIC_LINKER_SOFT_FLOAT DYNAMIC_LINKER_PREFIX "/lib/ld-linux.so.3"
+#define GLIBC_DYNAMIC_LINKER_HARD_FLOAT DYNAMIC_LINKER_PREFIX "/lib/ld-linux-armhf.so.3"
 #define GLIBC_DYNAMIC_LINKER_DEFAULT GLIBC_DYNAMIC_LINKER_SOFT_FLOAT
 
 #define GLIBC_DYNAMIC_LINKER \
Index: gcc-4.9.2/gcc/config/i386/linux.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/i386/linux.h
+++ gcc-4.9.2/gcc/config/i386/linux.h
@@ -20,4 +20,4 @@ along with GCC; see the file COPYING3.
 <http://www.gnu.org/licenses/>.  */
 
 #define GNU_USER_LINK_EMULATION "elf_i386"
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld-linux.so.2"
Index: gcc-4.9.2/gcc/config.in
===================================================================
--- gcc-4.9.2.orig/gcc/config.in
+++ gcc-4.9.2/gcc/config.in
@@ -39,6 +39,12 @@
 #endif
 
 
+/* root directory of the dynamic linker */
+#ifndef USED_FOR_TARGET
+#undef DYNAMIC_LINKER_PREFIX
+#endif
+
+
 /* Define if you want assertions enabled. This is a cheap check. */
 #ifndef USED_FOR_TARGET
 #undef ENABLE_ASSERT_CHECKING
Index: gcc-4.9.2/gcc/configure
===================================================================
--- gcc-4.9.2.orig/gcc/configure
+++ gcc-4.9.2/gcc/configure
@@ -894,6 +894,7 @@ enable_objc_gc
 with_dwarf2
 enable_shared
 with_native_system_header_dir
+with_dynamic_linker_prefix
 with_build_sysroot
 with_sysroot
 with_specs
@@ -1682,6 +1683,8 @@ Optional Packages:
   --with-native-system-header-dir=dir
                           use dir as the directory to look for standard
                           system header files in.  Defaults to /usr/include.
+  --with-dynamic-linker-prefix=dir
+                          use dir as the root directory of the dynamic linker.
   --with-build-sysroot=sysroot
                           use sysroot as the system root during the build
   --with-sysroot[=DIR]    search for usr/lib, usr/include, et al, within DIR
@@ -7154,6 +7157,22 @@ fi
 
 
 
+# Check whether --with-dynamic-linker-prefix was given.
+if test "${with_dynamic_linker_prefix+set}" = set; then :
+  withval=$with_dynamic_linker_prefix; if test x"$withval" != x ; then
+     DYNAMIC_LINKER_PREFIX="$withval"
+  fi
+else
+  DYNAMIC_LINKER_PREFIX=
+fi
+
+
+cat >>confdefs.h <<_ACEOF
+#define DYNAMIC_LINKER_PREFIX "$DYNAMIC_LINKER_PREFIX"
+_ACEOF
+
+
+
 # Check whether --with-build-sysroot was given.
 if test "${with_build_sysroot+set}" = set; then :
   withval=$with_build_sysroot; if test x"$withval" != x ; then
@@ -17936,7 +17955,7 @@ else
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<_LT_EOF
-#line 17939 "configure"
+#line 17958 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -18042,7 +18061,7 @@ else
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<_LT_EOF
-#line 18045 "configure"
+#line 18064 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
Index: gcc-4.9.2/gcc/configure.ac
===================================================================
--- gcc-4.9.2.orig/gcc/configure.ac
+++ gcc-4.9.2/gcc/configure.ac
@@ -751,6 +751,16 @@ AC_ARG_WITH([native-system-header-dir],
  configured_native_system_header_dir="${withval}"
 ], [configured_native_system_header_dir=])
 
+AC_ARG_WITH([dynamic-linker-prefix],
+  [AS_HELP_STRING([--with-dynamic-linker-prefix=dir],
+                  [use dir as the root directory of the dynamic linker.])],
+  [if test x"$withval" != x ; then
+     DYNAMIC_LINKER_PREFIX="$withval"
+  fi],
+  [DYNAMIC_LINKER_PREFIX=])
+AC_DEFINE_UNQUOTED(DYNAMIC_LINKER_PREFIX, "$DYNAMIC_LINKER_PREFIX",
+                   [root directory of the dynamic linker])
+
 AC_ARG_WITH(build-sysroot, 
   [AS_HELP_STRING([--with-build-sysroot=sysroot],
                   [use sysroot as the system root during the build])],
Index: gcc-4.9.2/gcc/config/aarch64/aarch64-linux.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/aarch64/aarch64-linux.h
+++ gcc-4.9.2/gcc/config/aarch64/aarch64-linux.h
@@ -21,7 +21,7 @@
 #ifndef GCC_AARCH64_LINUX_H
 #define GCC_AARCH64_LINUX_H
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-aarch64%{mbig-endian:_be}.so.1"
+#define GLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld-linux-aarch64%{mbig-endian:_be}.so.1"
 
 #define CPP_SPEC "%{pthread:-D_REENTRANT}"
 
Index: gcc-4.9.2/gcc/config/alpha/linux-elf.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/alpha/linux-elf.h
+++ gcc-4.9.2/gcc/config/alpha/linux-elf.h
@@ -24,7 +24,7 @@ along with GCC; see the file COPYING3.
 { "elf_dynamic_linker", ELF_DYNAMIC_LINKER },
 
 #define GLIBC_DYNAMIC_LINKER	"/lib/ld-linux.so.2"
-#define UCLIBC_DYNAMIC_LINKER "/lib/ld-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld-uClibc.so.0"
 #if DEFAULT_LIBC == LIBC_UCLIBC
 #define CHOOSE_DYNAMIC_LINKER(G, U) "%{mglibc:" G ";:" U "}"
 #elif DEFAULT_LIBC == LIBC_GLIBC
Index: gcc-4.9.2/gcc/config/arm/linux-elf.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/arm/linux-elf.h
+++ gcc-4.9.2/gcc/config/arm/linux-elf.h
@@ -62,7 +62,7 @@
 
 #define LIBGCC_SPEC "%{mfloat-abi=soft*:-lfloat} -lgcc"
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld-linux.so.2"
 
 #define LINUX_TARGET_LINK_SPEC  "%{h*} \
    %{static:-Bstatic} \
Index: gcc-4.9.2/gcc/config/c6x/uclinux-elf.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/c6x/uclinux-elf.h
+++ gcc-4.9.2/gcc/config/c6x/uclinux-elf.h
@@ -36,7 +36,7 @@
 #define STARTFILE_SPEC \
   "%{!shared:crt1%O%s} crti%O%s %{shared|pie:crtbeginS.o%s;:crtbegin.o%s}"
 
-#define UCLIBC_DYNAMIC_LINKER "/lib/ld-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld-uClibc.so.0"
 
 #undef LINK_SPEC
 #define LINK_SPEC ENDIAN_LINK_SPEC \
Index: gcc-4.9.2/gcc/config/cris/linux.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/cris/linux.h
+++ gcc-4.9.2/gcc/config/cris/linux.h
@@ -102,7 +102,7 @@ along with GCC; see the file COPYING3.
 #undef CRIS_DEFAULT_CPU_VERSION
 #define CRIS_DEFAULT_CPU_VERSION CRIS_CPU_NG
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld.so.1"
 
 #undef CRIS_LINK_SUBTARGET_SPEC
 #define CRIS_LINK_SUBTARGET_SPEC \
Index: gcc-4.9.2/gcc/config/freebsd-spec.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/freebsd-spec.h
+++ gcc-4.9.2/gcc/config/freebsd-spec.h
@@ -128,9 +128,9 @@ is built with the --enable-threads confi
 #endif
 
 #if FBSD_MAJOR < 6
-#define FBSD_DYNAMIC_LINKER "/usr/libexec/ld-elf.so.1"
+#define FBSD_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/usr/libexec/ld-elf.so.1"
 #else
-#define FBSD_DYNAMIC_LINKER "/libexec/ld-elf.so.1"
+#define FBSD_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/libexec/ld-elf.so.1"
 #endif
 
 /* NOTE: The freebsd-spec.h header is included also for various
Index: gcc-4.9.2/gcc/config/frv/linux.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/frv/linux.h
+++ gcc-4.9.2/gcc/config/frv/linux.h
@@ -34,7 +34,7 @@
 #define ENDFILE_SPEC \
   "%{shared|pie:crtendS.o%s;:crtend.o%s} crtn.o%s"
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld.so.1"
 
 #undef LINK_SPEC
 #define LINK_SPEC "\
Index: gcc-4.9.2/gcc/config/i386/gnu.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/i386/gnu.h
+++ gcc-4.9.2/gcc/config/i386/gnu.h
@@ -22,7 +22,7 @@ along with GCC.  If not, see <http://www
 #define GNU_USER_LINK_EMULATION "elf_i386"
 
 #undef GNU_USER_DYNAMIC_LINKER
-#define GNU_USER_DYNAMIC_LINKER "/lib/ld.so"
+#define GNU_USER_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld.so"
 
 #undef	STARTFILE_SPEC
 #if defined HAVE_LD_PIE
Index: gcc-4.9.2/gcc/config/i386/kfreebsd-gnu64.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/i386/kfreebsd-gnu64.h
+++ gcc-4.9.2/gcc/config/i386/kfreebsd-gnu64.h
@@ -22,6 +22,6 @@ along with GCC; see the file COPYING3.
 #define GNU_USER_LINK_EMULATION64 "elf_x86_64_fbsd"
 #define GNU_USER_LINK_EMULATIONX32 "elf32_x86_64_fbsd"
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
-#define GLIBC_DYNAMIC_LINKER64 "/lib/ld-kfreebsd-x86-64.so.1"
-#define GLIBC_DYNAMIC_LINKERX32 "/lib/ld-kfreebsd-x32.so.1"
+#define GLIBC_DYNAMIC_LINKER32 DYNAMIC_LINKER_PREFIX "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER64 DYNAMIC_LINKER_PREFIX "/lib/ld-kfreebsd-x86-64.so.1"
+#define GLIBC_DYNAMIC_LINKERX32 DYNAMIC_LINKER_PREFIX "/lib/ld-kfreebsd-x32.so.1"
Index: gcc-4.9.2/gcc/config/i386/kfreebsd-gnu.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/i386/kfreebsd-gnu.h
+++ gcc-4.9.2/gcc/config/i386/kfreebsd-gnu.h
@@ -19,4 +19,4 @@ along with GCC; see the file COPYING3.
 <http://www.gnu.org/licenses/>.  */
 
 #define GNU_USER_LINK_EMULATION "elf_i386_fbsd"
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld.so.1"
Index: gcc-4.9.2/gcc/config/ia64/linux.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/ia64/linux.h
+++ gcc-4.9.2/gcc/config/ia64/linux.h
@@ -55,7 +55,7 @@ do {						\
 /* Define this for shared library support because it isn't in the main
    linux.h file.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-ia64.so.2"
+#define GLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld-linux-ia64.so.2"
 
 #undef LINK_SPEC
 #define LINK_SPEC "\
Index: gcc-4.9.2/gcc/config/knetbsd-gnu.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/knetbsd-gnu.h
+++ gcc-4.9.2/gcc/config/knetbsd-gnu.h
@@ -32,4 +32,4 @@ along with GCC; see the file COPYING3.
 
 
 #undef GNU_USER_DYNAMIC_LINKER
-#define GNU_USER_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GNU_USER_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld.so.1"
Index: gcc-4.9.2/gcc/config/kopensolaris-gnu.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/kopensolaris-gnu.h
+++ gcc-4.9.2/gcc/config/kopensolaris-gnu.h
@@ -31,4 +31,4 @@ along with GCC; see the file COPYING3.
   while (0)
 
 #undef GNU_USER_DYNAMIC_LINKER
-#define GNU_USER_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GNU_USER_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld.so.1"
Index: gcc-4.9.2/gcc/config/linux.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/linux.h
+++ gcc-4.9.2/gcc/config/linux.h
@@ -73,11 +73,11 @@ see the files COPYING3 and COPYING.RUNTI
    GLIBC_DYNAMIC_LINKER must be defined for each target using them, or
    GLIBC_DYNAMIC_LINKER32 and GLIBC_DYNAMIC_LINKER64 for targets
    supporting both 32-bit and 64-bit compilation.  */
-#define UCLIBC_DYNAMIC_LINKER "/lib/ld-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld-uClibc.so.0"
 #define UCLIBC_DYNAMIC_LINKER32 "/lib/ld-uClibc.so.0"
 #define UCLIBC_DYNAMIC_LINKER64 "/lib/ld64-uClibc.so.0"
 #define UCLIBC_DYNAMIC_LINKERX32 "/lib/ldx32-uClibc.so.0"
-#define BIONIC_DYNAMIC_LINKER "/system/bin/linker"
+#define BIONIC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/system/bin/linker"
 #define BIONIC_DYNAMIC_LINKER32 "/system/bin/linker"
 #define BIONIC_DYNAMIC_LINKER64 "/system/bin/linker64"
 #define BIONIC_DYNAMIC_LINKERX32 "/system/bin/linkerx32"
Index: gcc-4.9.2/gcc/config/m32r/linux.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/m32r/linux.h
+++ gcc-4.9.2/gcc/config/m32r/linux.h
@@ -37,7 +37,7 @@
    When the -shared link option is used a final link is not being
    done.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld-linux.so.2"
 
 #undef	LINK_SPEC
 #if TARGET_LITTLE_ENDIAN
Index: gcc-4.9.2/gcc/config/m68k/linux.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/m68k/linux.h
+++ gcc-4.9.2/gcc/config/m68k/linux.h
@@ -71,7 +71,7 @@ along with GCC; see the file COPYING3.
    When the -shared link option is used a final link is not being
    done.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld.so.1"
 
 #undef LINK_SPEC
 #define LINK_SPEC "-m m68kelf %{shared} \
Index: gcc-4.9.2/gcc/config/microblaze/linux.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/microblaze/linux.h
+++ gcc-4.9.2/gcc/config/microblaze/linux.h
@@ -25,7 +25,7 @@
 #undef TLS_NEEDS_GOT
 #define TLS_NEEDS_GOT 1
 
-#define DYNAMIC_LINKER "/lib/ld.so.1"
+#define DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld.so.1"
 #undef  SUBTARGET_EXTRA_SPECS
 #define SUBTARGET_EXTRA_SPECS \
   { "dynamic_linker", DYNAMIC_LINKER }
Index: gcc-4.9.2/gcc/config/mips/linux64.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/mips/linux64.h
+++ gcc-4.9.2/gcc/config/mips/linux64.h
@@ -38,7 +38,7 @@ along with GCC; see the file COPYING3.
 #define UCLIBC_DYNAMIC_LINKERN32 \
   "%{mnan=2008:/lib32/ld-uClibc-mipsn8.so.0;:/lib32/ld-uClibc.so.0}"
 
-#define BIONIC_DYNAMIC_LINKERN32 "/system/bin/linker32"
+#define BIONIC_DYNAMIC_LINKERN32 DYNAMIC_LINKER_PREFIX "/system/bin/linker32"
 #define GNU_USER_DYNAMIC_LINKERN32 \
   CHOOSE_DYNAMIC_LINKER (GLIBC_DYNAMIC_LINKERN32, UCLIBC_DYNAMIC_LINKERN32, \
 			 BIONIC_DYNAMIC_LINKERN32)
Index: gcc-4.9.2/gcc/config/mn10300/linux.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/mn10300/linux.h
+++ gcc-4.9.2/gcc/config/mn10300/linux.h
@@ -32,7 +32,7 @@
 #undef  ASM_SPEC
 #define ASM_SPEC ""
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld.so.1"
 
 #undef  LINK_SPEC
 #define LINK_SPEC "%{mrelax:--relax} %{shared:-shared} \
Index: gcc-4.9.2/gcc/config/nios2/linux.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/nios2/linux.h
+++ gcc-4.9.2/gcc/config/nios2/linux.h
@@ -26,7 +26,7 @@
     }                                           \
   while (0)
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-nios2.so.1"
+#define GLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld-linux-nios2.so.1"
 
 #undef LINK_SPEC
 #define LINK_SPEC LINK_SPEC_ENDIAN \
Index: gcc-4.9.2/gcc/config/pa/pa-linux.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/pa/pa-linux.h
+++ gcc-4.9.2/gcc/config/pa/pa-linux.h
@@ -40,7 +40,7 @@ along with GCC; see the file COPYING3.
 /* Define this for shared library support because it isn't in the main
    linux.h file.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld.so.1"
 
 #undef LINK_SPEC
 #define LINK_SPEC "\
Index: gcc-4.9.2/gcc/config/rs6000/freebsd64.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/rs6000/freebsd64.h
+++ gcc-4.9.2/gcc/config/rs6000/freebsd64.h
@@ -154,8 +154,8 @@ extern int dot_symbols;
   { "link_os_freebsd_spec32",	LINK_OS_FREEBSD_SPEC32 },     		\
   { "link_os_freebsd_spec64",	LINK_OS_FREEBSD_SPEC64 },
 
-#define FREEBSD_DYNAMIC_LINKER32 "/libexec/ld-elf32.so.1"
-#define FREEBSD_DYNAMIC_LINKER64 "/libexec/ld-elf.so.1"
+#define FREEBSD_DYNAMIC_LINKER32 DYNAMIC_LINKER_PREFIX "/libexec/ld-elf32.so.1"
+#define FREEBSD_DYNAMIC_LINKER64 DYNAMIC_LINKER_PREFIX "/libexec/ld-elf.so.1"
 
 #define LINK_OS_FREEBSD_SPEC_DEF32 "\
   %{p:%nconsider using `-pg' instead of `-p' with gprof(1)} \
Index: gcc-4.9.2/gcc/config/rs6000/linux64.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/rs6000/linux64.h
+++ gcc-4.9.2/gcc/config/rs6000/linux64.h
@@ -367,14 +367,14 @@ extern int dot_symbols;
 #undef	LINK_OS_DEFAULT_SPEC
 #define LINK_OS_DEFAULT_SPEC "%(link_os_linux)"
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER32 DYNAMIC_LINKER_PREFIX "/lib/ld.so.1"
 #ifdef LINUX64_DEFAULT_ABI_ELFv2
-#define GLIBC_DYNAMIC_LINKER64 "%{mabi=elfv1:/lib64/ld64.so.1;:/lib64/ld64.so.2}"
+#define GLIBC_DYNAMIC_LINKER64 DYNAMIC_LINKER_PREFIX "%{mabi=elfv1:/lib64/ld64.so.1;:/lib64/ld64.so.2}"
 #else
-#define GLIBC_DYNAMIC_LINKER64 "%{mabi=elfv2:/lib64/ld64.so.2;:/lib64/ld64.so.1}"
+#define GLIBC_DYNAMIC_LINKER64 DYNAMIC_LINKER_PREFIX "%{mabi=elfv2:/lib64/ld64.so.2;:/lib64/ld64.so.1}"
 #endif
-#define UCLIBC_DYNAMIC_LINKER32 "/lib/ld-uClibc.so.0"
-#define UCLIBC_DYNAMIC_LINKER64 "/lib/ld64-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER32 DYNAMIC_LINKER_PREFIX "/lib/ld-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER64 DYNAMIC_LINKER_PREFIX "/lib/ld64-uClibc.so.0"
 #if DEFAULT_LIBC == LIBC_UCLIBC
 #define CHOOSE_DYNAMIC_LINKER(G, U) "%{mglibc:" G ";:" U "}"
 #elif DEFAULT_LIBC == LIBC_GLIBC
Index: gcc-4.9.2/gcc/config/rs6000/sysv4.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/rs6000/sysv4.h
+++ gcc-4.9.2/gcc/config/rs6000/sysv4.h
@@ -761,8 +761,8 @@ ENDIAN_SELECT(" -mbig", " -mlittle", DEF
 
 #define LINK_START_LINUX_SPEC ""
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
-#define UCLIBC_DYNAMIC_LINKER "/lib/ld-uClibc.so.0"
+#define GLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld.so.1"
+#define UCLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld-uClibc.so.0"
 #if DEFAULT_LIBC == LIBC_UCLIBC
 #define CHOOSE_DYNAMIC_LINKER(G, U) "%{mglibc:" G ";:" U "}"
 #elif !defined (DEFAULT_LIBC) || DEFAULT_LIBC == LIBC_GLIBC
Index: gcc-4.9.2/gcc/config/s390/linux.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/s390/linux.h
+++ gcc-4.9.2/gcc/config/s390/linux.h
@@ -60,8 +60,8 @@ along with GCC; see the file COPYING3.
 #define MULTILIB_DEFAULTS { "m31" }
 #endif
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
-#define GLIBC_DYNAMIC_LINKER64 "/lib/ld64.so.1"
+#define GLIBC_DYNAMIC_LINKER32 DYNAMIC_LINKER_PREFIX "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER64 DYNAMIC_LINKER_PREFIX "/lib/ld64.so.1"
 
 #undef  LINK_SPEC
 #define LINK_SPEC \
Index: gcc-4.9.2/gcc/config/sh/linux.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/sh/linux.h
+++ gcc-4.9.2/gcc/config/sh/linux.h
@@ -43,7 +43,7 @@ along with GCC; see the file COPYING3.
 
 #define TARGET_ASM_FILE_END file_end_indicate_exec_stack
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld-linux.so.2"
 
 #undef SUBTARGET_LINK_EMUL_SUFFIX
 #define SUBTARGET_LINK_EMUL_SUFFIX "_linux"
Index: gcc-4.9.2/gcc/config/sparc/linux64.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/sparc/linux64.h
+++ gcc-4.9.2/gcc/config/sparc/linux64.h
@@ -92,8 +92,8 @@ along with GCC; see the file COPYING3.
    When the -shared link option is used a final link is not being
    done.  */
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER32 DYNAMIC_LINKER_PREFIX "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER64 DYNAMIC_LINKER_PREFIX "/lib64/ld-linux.so.2"
 
 #ifdef SPARC_BI_ARCH
 
Index: gcc-4.9.2/gcc/config/sparc/linux.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/sparc/linux.h
+++ gcc-4.9.2/gcc/config/sparc/linux.h
@@ -83,7 +83,7 @@ extern const char *host_detect_local_cpu
    When the -shared link option is used a final link is not being
    done.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld-linux.so.2"
 
 #undef  LINK_SPEC
 #define LINK_SPEC "-m elf32_sparc %{shared:-shared} \
Index: gcc-4.9.2/gcc/config/xtensa/linux.h
===================================================================
--- gcc-4.9.2.orig/gcc/config/xtensa/linux.h
+++ gcc-4.9.2/gcc/config/xtensa/linux.h
@@ -44,7 +44,7 @@ along with GCC; see the file COPYING3.
   %{mlongcalls:--longcalls} \
   %{mno-longcalls:--no-longcalls}"
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER DYNAMIC_LINKER_PREFIX "/lib/ld.so.1"
 
 #undef LINK_SPEC
 #define LINK_SPEC \
