Index: work/glibc-2.17/sysdeps/generic/paths.h
===================================================================
--- work.orig/glibc-2.17/sysdeps/generic/paths.h
+++ work/glibc-2.17/sysdeps/generic/paths.h
@@ -33,43 +33,43 @@
 #define	_PATHS_H_
 
 /* Default search path. */
-#define	_PATH_DEFPATH	"/usr/bin:/bin"
+#define	_PATH_DEFPATH	"@GENTOO_PORTAGE_EPREFIX@/usr/bin:@GENTOO_PORTAGE_EPREFIX@/bin"
 /* All standard utilities path. */
 #define	_PATH_STDPATH \
-	"/usr/bin:/bin:/usr/sbin:/sbin"
+	"@GENTOO_PORTAGE_EPREFIX@/usr/bin:@GENTOO_PORTAGE_EPREFIX@/bin:@GENTOO_PORTAGE_EPREFIX@/usr/sbin:@GENTOO_PORTAGE_EPREFIX@/sbin"
 
-#define	_PATH_BSHELL	"/bin/sh"
+#define	_PATH_BSHELL	"@GENTOO_PORTAGE_EPREFIX@/bin/sh"
 #define	_PATH_CONSOLE	"/dev/console"
-#define	_PATH_CSHELL	"/bin/csh"
-#define	_PATH_DEVDB	"/var/run/dev.db"
+#define	_PATH_CSHELL	"@GENTOO_PORTAGE_EPREFIX@/bin/csh"
+#define	_PATH_DEVDB	"@GENTOO_PORTAGE_EPREFIX@/var/run/dev.db"
 #define	_PATH_DEVNULL	"/dev/null"
 #define	_PATH_DRUM	"/dev/drum"
-#define	_PATH_GSHADOW	"/etc/gshadow"
+#define	_PATH_GSHADOW	"@GENTOO_PORTAGE_EPREFIX@/etc/gshadow"
 #define	_PATH_KMEM	"/dev/kmem"
-#define	_PATH_LASTLOG	"/var/log/lastlog"
+#define	_PATH_LASTLOG	"@GENTOO_PORTAGE_EPREFIX@/var/log/lastlog"
 #define	_PATH_MAILDIR	"/var/mail"
-#define	_PATH_MAN	"/usr/share/man"
+#define	_PATH_MAN	"@GENTOO_PORTAGE_EPREFIX@/usr/share/man"
 #define	_PATH_MEM	"/dev/mem"
-#define	_PATH_MNTTAB	"/etc/fstab"
-#define	_PATH_MOUNTED	"/var/run/mtab"
-#define	_PATH_NOLOGIN	"/etc/nologin"
-#define	_PATH_PRESERVE	"/var/lib"
-#define	_PATH_RWHODIR	"/var/spool/rwho"
-#define	_PATH_SENDMAIL	"/usr/sbin/sendmail"
-#define	_PATH_SHADOW	"/etc/shadow"
-#define	_PATH_SHELLS	"/etc/shells"
+#define	_PATH_MNTTAB	"@GENTOO_PORTAGE_EPREFIX@/etc/fstab"
+#define	_PATH_MOUNTED	"@GENTOO_PORTAGE_EPREFIX@/var/run/mtab"
+#define	_PATH_NOLOGIN	"@GENTOO_PORTAGE_EPREFIX@/etc/nologin"
+#define	_PATH_PRESERVE	"@GENTOO_PORTAGE_EPREFIX@/var/lib"
+#define	_PATH_RWHODIR	"@GENTOO_PORTAGE_EPREFIX@/var/spool/rwho"
+#define	_PATH_SENDMAIL	"@GENTOO_PORTAGE_EPREFIX@/usr/sbin/sendmail"
+#define	_PATH_SHADOW	"@GENTOO_PORTAGE_EPREFIX@/etc/shadow"
+#define	_PATH_SHELLS	"@GENTOO_PORTAGE_EPREFIX@/etc/shells"
 #define	_PATH_TTY	"/dev/tty"
 #define	_PATH_UNIX	"/vmunix"
-#define	_PATH_UTMP	"/var/run/utmp"
-#define	_PATH_UTMP_DB	"/var/run/utmp.db"
-#define	_PATH_VI	"/usr/bin/vi"
+#define	_PATH_UTMP	"@GENTOO_PORTAGE_EPREFIX@/var/run/utmp"
+#define	_PATH_UTMP_DB	"@GENTOO_PORTAGE_EPREFIX@/var/run/utmp.db"
+#define	_PATH_VI	"@GENTOO_PORTAGE_EPREFIX@/usr/bin/vi"
 #define	_PATH_WTMP	"/var/log/wtmp"
 
 /* Provide trailing slash, since mostly used for building pathnames. */
 #define	_PATH_DEV	"/dev/"
 #define	_PATH_TMP	"/tmp/"
-#define	_PATH_VARDB	"/var/db/"
-#define	_PATH_VARRUN	"/var/run/"
+#define	_PATH_VARDB	"@GENTOO_PORTAGE_EPREFIX@/var/db/"
+#define	_PATH_VARRUN	"@GENTOO_PORTAGE_EPREFIX@/var/run/"
 #define	_PATH_VARTMP	"/var/tmp/"
 
 #endif /* !_PATHS_H_ */
Index: work/glibc-2.17/sysdeps/unix/sysv/linux/paths.h
===================================================================
--- work.orig/glibc-2.17/sysdeps/unix/sysv/linux/paths.h
+++ work/glibc-2.17/sysdeps/unix/sysv/linux/paths.h
@@ -33,43 +33,43 @@
 #define	_PATHS_H_
 
 /* Default search path. */
-#define	_PATH_DEFPATH	"/usr/bin:/bin"
+#define	_PATH_DEFPATH	"@GENTOO_PORTAGE_EPREFIX@/usr/bin:@GENTOO_PORTAGE_EPREFIX@/bin"
 /* All standard utilities path. */
 #define	_PATH_STDPATH \
-	"/usr/bin:/bin:/usr/sbin:/sbin"
+	"@GENTOO_PORTAGE_EPREFIX@/usr/bin:@GENTOO_PORTAGE_EPREFIX@/bin:@GENTOO_PORTAGE_EPREFIX@/usr/sbin:@GENTOO_PORTAGE_EPREFIX@/sbin"
 
-#define	_PATH_BSHELL	"/bin/sh"
+#define	_PATH_BSHELL	"@GENTOO_PORTAGE_EPREFIX@/bin/sh"
 #define	_PATH_CONSOLE	"/dev/console"
-#define	_PATH_CSHELL	"/bin/csh"
-#define	_PATH_DEVDB	"/var/run/dev.db"
+#define	_PATH_CSHELL	"@GENTOO_PORTAGE_EPREFIX@/bin/csh"
+#define	_PATH_DEVDB	"@GENTOO_PORTAGE_EPREFIX@/var/run/dev.db"
 #define	_PATH_DEVNULL	"/dev/null"
 #define	_PATH_DRUM	"/dev/drum"
-#define	_PATH_GSHADOW	"/etc/gshadow"
+#define	_PATH_GSHADOW	"@GENTOO_PORTAGE_EPREFIX@/etc/gshadow"
 #define	_PATH_KLOG	"/proc/kmsg"
 #define	_PATH_KMEM	"/dev/kmem"
 #define	_PATH_LASTLOG	"/var/log/lastlog"
 #define	_PATH_MAILDIR	"/var/mail"
-#define	_PATH_MAN	"/usr/share/man"
+#define	_PATH_MAN	"@GENTOO_PORTAGE_EPREFIX@/usr/share/man"
 #define	_PATH_MEM	"/dev/mem"
-#define	_PATH_MNTTAB	"/etc/fstab"
-#define	_PATH_MOUNTED	"/etc/mtab"
-#define	_PATH_NOLOGIN	"/etc/nologin"
-#define	_PATH_PRESERVE	"/var/lib"
-#define	_PATH_RWHODIR	"/var/spool/rwho"
-#define	_PATH_SENDMAIL	"/usr/sbin/sendmail"
-#define	_PATH_SHADOW	"/etc/shadow"
-#define	_PATH_SHELLS	"/etc/shells"
+#define	_PATH_MNTTAB	"@GENTOO_PORTAGE_EPREFIX@/etc/fstab"
+#define	_PATH_MOUNTED	"@GENTOO_PORTAGE_EPREFIX@/etc/mtab"
+#define	_PATH_NOLOGIN	"@GENTOO_PORTAGE_EPREFIX@/etc/nologin"
+#define	_PATH_PRESERVE	"@GENTOO_PORTAGE_EPREFIX@/var/lib"
+#define	_PATH_RWHODIR	"@GENTOO_PORTAGE_EPREFIX@/var/spool/rwho"
+#define	_PATH_SENDMAIL	"@GENTOO_PORTAGE_EPREFIX@/usr/sbin/sendmail"
+#define	_PATH_SHADOW	"@GENTOO_PORTAGE_EPREFIX@/etc/shadow"
+#define	_PATH_SHELLS	"@GENTOO_PORTAGE_EPREFIX@/etc/shells"
 #define	_PATH_TTY	"/dev/tty"
 #define	_PATH_UNIX	"/boot/vmlinux"
-#define	_PATH_UTMP	"/var/run/utmp"
-#define	_PATH_VI	"/usr/bin/vi"
+#define	_PATH_UTMP	"@GENTOO_PORTAGE_EPREFIX@/var/run/utmp"
+#define	_PATH_VI	"@GENTOO_PORTAGE_EPREFIX@/usr/bin/vi"
 #define	_PATH_WTMP	"/var/log/wtmp"
 
 /* Provide trailing slash, since mostly used for building pathnames. */
 #define	_PATH_DEV	"/dev/"
 #define	_PATH_TMP	"/tmp/"
-#define	_PATH_VARDB	"/var/db/"
-#define	_PATH_VARRUN	"/var/run/"
+#define	_PATH_VARDB	"@GENTOO_PORTAGE_EPREFIX@/var/db/"
+#define	_PATH_VARRUN	"@GENTOO_PORTAGE_EPREFIX@/var/run/"
 #define	_PATH_VARTMP	"/var/tmp/"
 
 #endif /* !_PATHS_H_ */
Index: work/glibc-2.17/sysdeps/posix/system.c
===================================================================
--- work.orig/glibc-2.17/sysdeps/posix/system.c
+++ work/glibc-2.17/sysdeps/posix/system.c
@@ -26,7 +26,7 @@
 #include <sysdep-cancel.h>
 
 
-#define	SHELL_PATH	"/bin/sh"	/* Path of the shell.  */
+#define	SHELL_PATH	"@GENTOO_PORTAGE_EPREFIX@/bin/sh"	/* Path of the shell.  */
 #define	SHELL_NAME	"sh"		/* Name to give it.  */
 
 
Index: work/glibc-2.17/libio/iopopen.c
===================================================================
--- work.orig/glibc-2.17/libio/iopopen.c
+++ work/glibc-2.17/libio/iopopen.c
@@ -222,7 +222,7 @@ _IO_new_proc_open (fp, command, mode)
 	    _IO_close (fd);
 	}
 
-      _IO_execl ("/bin/sh", "sh", "-c", command, (char *) 0);
+      _IO_execl ("@GENTOO_PORTAGE_EPREFIX@/bin/sh", "sh", "-c", command, (char *) 0);
       _IO__exit (127);
     }
   _IO_close (child_end);
