--- test/Makefile.in	2012-03-05 13:26:51.000000000 +0900
+++ tmp/Makefile.in	2012-03-28 11:00:37.086315000 +0900
@@ -126,7 +126,7 @@ am__DEPENDENCIES_1 =
 @ENABLE_UNIT_TESTS_TRUE@@XORG_TRUE@	$(top_builddir)/hw/xfree86/i2c/libi2c.la \
 @ENABLE_UNIT_TESTS_TRUE@@XORG_TRUE@	$(top_builddir)/hw/xfree86/dixmods/libxorgxkb.la
 @ENABLE_UNIT_TESTS_TRUE@@XORG_FALSE@nodist_libxservertest_la_OBJECTS =  \
-@ENABLE_UNIT_TESTS_TRUE@@XORG_FALSE@	ddxstubs.lo miinitext.lo \
+@ENABLE_UNIT_TESTS_TRUE@@XORG_TRUE@	ddxstubs.lo miinitext.lo \
 @ENABLE_UNIT_TESTS_TRUE@@XORG_FALSE@	dpmsstubs.lo stubs.lo
 @ENABLE_UNIT_TESTS_TRUE@@XORG_TRUE@nodist_libxservertest_la_OBJECTS =  \
 @ENABLE_UNIT_TESTS_TRUE@@XORG_TRUE@	sdksyms.lo
@@ -598,7 +598,7 @@ top_srcdir = @top_srcdir@
 @ENABLE_UNIT_TESTS_TRUE@	$(am__append_9) $(am__append_10) \
 @ENABLE_UNIT_TESTS_TRUE@	$(am__append_11) $(am__append_13)
 @ENABLE_UNIT_TESTS_TRUE@@XORG_FALSE@nodist_libxservertest_la_SOURCES = \
-@ENABLE_UNIT_TESTS_TRUE@@XORG_FALSE@            ddxstubs.c \
+@ENABLE_UNIT_TESTS_TRUE@@XORG_TRUE@            ddxstubs.c \
 @ENABLE_UNIT_TESTS_TRUE@@XORG_FALSE@            $(top_srcdir)/mi/miinitext.c \
 @ENABLE_UNIT_TESTS_TRUE@@XORG_FALSE@            $(top_srcdir)/Xext/dpmsstubs.c \
 @ENABLE_UNIT_TESTS_TRUE@@XORG_FALSE@            $(top_srcdir)/Xi/stubs.c
