--- mk/sys.mk	2012-04-24 09:11:02.000000000 +0900
+++ mk/sys.mk.new	2012-04-24 10:32:16.000000000 +0900
@@ -11,7 +11,7 @@ SED?=			sed
 SH=			/bin/sh
 
 PREFIX?=
-_UPREFIX_SH=		case "${PREFIX}" in "") echo /usr;; *) echo "${PREFIX}";; esac
+_UPREFIX_SH=		case "${PREFIX}" in "") echo /usr;; *) echo "${PREFIX}"/usr;; esac
 _UPREFIX:=		$(shell ${_UPREFIX_SH})
 UPREFIX=		${_UPREFIX}
 LOCAL_PREFIX=		/usr/local
