Index: toolchain-prefix-wrapper-0.3.1723/ld/create-ldwrapper.in
===================================================================
--- toolchain-prefix-wrapper-0.3.1723.orig/ld/create-ldwrapper.in
+++ toolchain-prefix-wrapper-0.3.1723/ld/create-ldwrapper.in
@@ -78,13 +78,6 @@ doit() {
 	) > "${ctargetstem}".c || return 1
 	umask ${oldumask} || return 1
 
-	if [[ -z ${ROOT%/} && ${TARGET} == ${HOST} && ${EPREFIX} == ${BPREFIX} ]]; then
-		# currently installed ld-wrapper config may be out-of-date,
-		# so configure it by environment.
-		export BINUTILS_CONFIG_LD="${EPREFIX}/${BINPATH}/${WRAPPERNAME}"
-		export BINUTILS_CONFIG_LDTARGET="${HOST}"
-		export BINUTILS_CONFIG_LDPLUGINS="${plugins}"
-	fi
 	rm -f "${OUTPUT}.new" > /dev/null || return 1
 	${CC:-${HOST_CC}} ${CFLAGS} -o "${OUTPUT}.new" \
 		"${ctargetstem}".c "-I${WRAPPERSOURCE}" \
